Makefile

```diff
@@ -1,10 +1,43 @@
-bin: deps
+CLASS_PATH = .
+CFLAGS = -O2 -fPIC
+DIR = p2pclient/java
+JAVA_INCLUDES = -I$(JAVA_HOME)/include/darwin -I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/linux
+vpath %.class $(CLASS_PATH)
+
+.DEFAULT_GOAL := go-daemon
+
+java-daemon: libp2pd.jnilib
+
+libp2pd.jnilib : libp2pd.o
+	ld -dylib -flat_namespace -undefined suppress -macosx_version_min 10.13.4 -o $(DIR)/$@ $(DIR)/*.o -L$(DIR) -lp2pd
```

Comment:
can we find a solution that works on linux as well? Alternate, have explicit targets that call the rules with the macosx name.

Reply:
Is there a specific reason that `macosx_version_min 10.13.4` is set? I have upgraded to 10.14 so extra warnings showed up when I run this.

Reply:
Sorry for the stupid question(I only compiled shared libs in linux): is `libp2pd.jnilib` compiled with `fPIC`? I checked there is a GOT section, but not so sure that it means it is compiled as PIC. Thank you!

Reply:
> Sorry for the stupid question(I only compiled shared libs in linux): is libp2pd.jnilib compiled with fPIC? I checked there is a GOT section, but not so sure that it means it is compiled as PIC. Thank you!

It's not a stupid question. I had to be careful about how I ordered the build steps. If you look at the libp2p.o target you will see that I use $CFLAGS var for the gcc options. I defined those at the top of the file and -fPIC is one of them

Reply:
> can we find a solution that works on linux as well? Alternate, have explicit targets that call the rules with the macosx name.

@vyzo I def plan on doing that.  i will have the Makefile inspect uname to determine what OS to build for